frontend/src/components/MapView/Layers/AdminLevelDataLayer/index.tsx

```diff
@@ -120,7 +124,7 @@ const AdminLevelDataLayers = ({
 
         // load unique boundary only once
         // to avoid double loading which proven to be performance issue
-        if (!isLayerOnView(map, boundaryId)) {
+        if (!isLayerOnView(map?.getMap(), boundaryId)) {
```

Comment:
would it be worth moving the getMap call inside isLayerOnView?

Reply:
changed it so `getMap()` happens in the selector. tested it still works fine